vars/neoDeploy.groovy

```diff
@@ -216,11 +231,80 @@ void call(parameters = [:]) {
                 lock("$STEP_NAME:${neoCommandHelper.resourceLock()}") {
                     deploy(script, configuration, neoCommandHelper, configuration.dockerImage, deployMode)
                 }
+                if(configuration.neo.invalidateCache == true) {
+                    inValidateCache(configuration)
+                }
             }
         }
     }
 }
 
+private inValidateCache(configuration){
+    def account = configuration.neo.account
+    def host = configuration.neo.host
+
+    withCredentials([usernamePassword(
+        credentialsId: configuration.neo.oauthCredentialId,
+        passwordVariable: 'OAUTH_NEO_CLIENT_SECRET',
+        usernameVariable: 'OAUTH_NEO_CLIENT_ID')]) {
+        def bearerTokenResponse = sh(
+            script: """#!/bin/bash
+                 curl -X POST -u "${OAUTH_NEO_CLIENT_ID}:${OAUTH_NEO_CLIENT_SECRET}" \
+                 --fail \
+           \"https://oauthasservices-${account}.${host}/oauth2/api/v1/token?grant_type=client_credentials&scope=write,read\"
+                """,
+                returnStdout: true
+            )
+        def fetchBearerTokenJson = readJSON  text: bearerTokenResponse
+
+        def bearerToken = fetchBearerTokenJson.access_token
+
+        echo "Retrieved bearer token."
+
+        def xcsrfTokenResponse = sh(
+            script: """#!/bin/bash
+                              curl -i -L \
+                              -c 'cookies.jar' \
+                              -H 'X-CSRF-Token: Fetch' \
+                              -H "Authorization: Bearer ${bearerToken}" \
+                              --fail \
+                           \"https://sandboxportal-${account}.${host}/fiori/api/v1/csrf\"
+                    """, returnStdout: true)
+
+        def xcsrfTokenHeaderMatcher=xcsrfTokenResponse =~ /(?m)^X-CSRF-Token: ([0-9A-Z]*)$/
+        def xcsrfToken = xcsrfTokenHeaderMatcher[0][1]
```

Comment:
This can lead to serialisation issues as we have seen. One option is to extract this to a method and annotate the method with @NonCPS annotation. Alternatively I have successfully tested it with `readProperties`, as these are colon separated, as also Java properties can be:
```suggestion
        def responseMap = readProperties text: xcsrfTokenResponse
        def xcsrfToken = responseMap["X-CSRF-Token"]
```